dbrequests/database.py

```diff
@@ -70,7 +71,7 @@ def get_connection(self):
         if not self.open:
             raise exc.ResourceClosedError('Database closed.')
 
-        return Connection(self._engine.connect())
+        return self.connection_class(self._engine.connect())
```

Comment:
How would we deal with the situation in that a Connection constructor might require additional arguments. Currently we would require to add arguments to the methods, right? That, I think, is okay, as long as these arguments are not required during init of the Connection. Opening the connection happens outside of the Connection class itself, anyways. Is this an issue?

Reply:
If initializing with additional arguments ist necessary, one could just overwrite get_connection() in the child database class.

The alternative would be to add generic arguments to the initialization here and controll their validity with inspect, but this would obfuscate the code more than necessary, imho.

Reply:
Yes, sure. We can override that method. These OO principles are sometimes a bit alien to me. I think then we have all the flexibility we need!